projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { TabPanel } from '@wordpress/components';
+import { useCallback } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { MY_JETPACK_SECTION_OVERVIEW } from './constants';
+import styles from './styles.module.scss';
+import { TabContent } from './tab-content';
+import { MyJetpackSection } from './types';
+import { getMyJetpackSections, isValidMyJetpackSection } from './utils';
+
+/**
+ * My Jetpack Tab panel component.
+ *
+ * @return The rendered component.
+ */
+export function MyJetpackTabPanel() {
+	const params = useParams();
+	const navigate = useNavigate();
+
+	const onTabSelect = useCallback(
+		( tabName: string ) => {
+			if ( tabName !== params.section ) {
+				navigate( `/${ tabName }` );
+			}
+		},
+		[ navigate, params.section ]
+	);
+
+	const tabRenderer = useCallback( ( tab: { name: MyJetpackSection } ) => {
+		return <TabContent name={ tab.name } />;
+	}, [] );
+
+	// If the tab is not valid, use the default one.
+	const initialTab = isValidMyJetpackSection( params.section )
+		? params.section
+		: MY_JETPACK_SECTION_OVERVIEW;
```

Comment:
I believe we could have this value behind `useState( () => return initialTab and maybe do replaceState() )`

Reply:
Why? The URL should reflect immediately because `TabPanel` calls that `onSelect` with the default value and it in turn updates the URL accordingly. We don't need to manage any state here.